test/Staking.test.js

```diff
@@ -0,0 +1,299 @@
+        await time.increase(time.duration.seconds(60));
+        let balanceBefore = await instanceToken.balanceOf(acc3);
+        expect(balanceBefore).to.be.bignumber.equal(ether('0'));
+        let tx = await instanceStaking.unStake({ from: acc3 });
+        let balanceAfter = await instanceToken.balanceOf(acc3);
+        expectEvent(tx, "UnStake", { investor: acc3, amount: balanceAfter });
+        expect(Number(balanceAfter)).to.be.closeTo(Number(ether('220')), Number(ether('0.00008')));
+      });
+      it("Stake by acc2 false - Error : staking period has end", async () => {
+        await expectRevert(instanceStaking.stake(ether('50'), { from: acc2 }), "Error : staking period has end");
+      });
+      it("UnStake by acc4 after 365 days - done", async () => {
+        await time.increase(time.duration.days(235));
+        let balanceBefore = await instanceToken.balanceOf(acc4);
+        expect(balanceBefore).to.be.bignumber.equal(ether('0'));
+        let tx = await instanceStaking.unStake({ from: acc4 });
+        let balanceAfter = await instanceToken.balanceOf(acc4);
+        expectEvent(tx, "UnStake", { investor: acc4, amount: balanceAfter });
+        expect(Number(balanceAfter)).to.be.closeTo(Number(ether('330')), Number(ether('0.00008')));
+      });
+      it("UnStake by acc5 after 365 days - done", async () => {
+        let balanceBefore = await instanceToken.balanceOf(acc5);
+        expect(balanceBefore).to.be.bignumber.equal(ether('0'));
+        let tx = await instanceStaking.unStake({ from: acc5 });
+        let balanceAfter = await instanceToken.balanceOf(acc5);
+        expectEvent(tx, "UnStake", { investor: acc5, amount: balanceAfter });
+        expect(Number(balanceAfter)).to.be.closeTo(Number(ether('440')), Number(ether('0.00009')));
+      });
+      it("UnStake by acc6 after 365 days - done", async () => {
+        let balanceBefore = await instanceToken.balanceOf(acc6);
+        expect(balanceBefore).to.be.bignumber.equal(ether('0'));
+        let tx = await instanceStaking.unStake({ from: acc6 });
+        let balanceAfter = await instanceToken.balanceOf(acc6);
+        expectEvent(tx, "UnStake", { investor: acc6, amount: balanceAfter });
+        expect(Number(balanceAfter)).to.be.closeTo(Number(ether('532.1918')), Number(ether('0.00009')));
+      });
+      it("UnStake by acc2 in second time after 235 days - done", async () => {
+        let balanceBefore = await instanceToken.balanceOf(acc2);
+        expect(Number(balanceBefore)).to.be.closeTo(Number(ether('0')), Number(ether('0.008')));
+        let tx = await instanceStaking.unStake({ from: acc2 });
+        let balanceAfter = await instanceToken.balanceOf(acc2);
+        expectEvent(tx, "UnStake", { investor: acc2, amount: balanceAfter });
+        //before third stake , un stake value = 108.5753
+        expect(Number(balanceAfter)).to.be.closeTo(Number(ether('108.7041')), Number(ether('0.00008')));
+      });
```

Comment:
you also should add more test for check rights calculation, 

also check with another setRewards settings